mutiny-core/src/lib.rs

```diff
@@ -1369,39 +1429,94 @@ impl<S: MutinyStorage> MutinyWallet<S> {
-            })
-            .unwrap_or_default();
+    pub fn get_activity(
+        &self,
+        limit: Option<usize>,
+        offset: Option<usize>,
+    ) -> Result<Vec<ActivityItem>, MutinyError> {
+        let index = {
+            let index = self.storage.activity_index();
+            let vec = index.try_read()?.clone().into_iter().collect_vec();
+
+            match (limit, offset) {
+                (None, None) => vec,
+                (Some(limit), Some(offset)) => {
+                    let start = offset;
+                    let end = start.saturating_add(limit).min(vec.len());
+                    vec[start..end].to_vec()
+                }
+                (None, Some(offset)) => vec[offset..].to_vec(),
+                (Some(limit), None) => vec[..limit].to_vec(),
+            }
+        };
 
-        // Get activities from node manager
-        let (closures, onchain) = self.node_manager.get_activity().await?;
+        let labels_map = self.storage.get_invoice_labels()?;
 
-        let mut activities = Vec::with_capacity(lightning.len() + onchain.len() + closures.len());
-        for ln in lightning {
-            // Only show paid and in-flight invoices
-            match ln.status {
-                HTLCStatus::Succeeded | HTLCStatus::InFlight => {
-                    activities.push(ActivityItem::Lightning(Box::new(ln)));
+        let mut activities = Vec::with_capacity(index.len());
+        for item in index {
+            if item.key.starts_with(PAYMENT_INBOUND_PREFIX_KEY) {
+                if let Some(mutiny_invoice) =
+                    self.get_invoice_internal(&item.key, true, &labels_map)?
+                {
+                    activities.push(ActivityItem::Lightning(Box::new(mutiny_invoice)));
+                }
+            } else if item.key.starts_with(PAYMENT_OUTBOUND_PREFIX_KEY) {
+                if let Some(mutiny_invoice) =
+                    self.get_invoice_internal(&item.key, false, &labels_map)?
+                {
+                    activities.push(ActivityItem::Lightning(Box::new(mutiny_invoice)));
+                }
+            } else if item.key.starts_with(CHANNEL_CLOSURE_PREFIX) {
+                if let Some(mut closure) = self.storage.get_data::<ChannelClosure>(&item.key)? {
+                    if closure.user_channel_id.is_none() {
+                        // convert keys to u128
+                        let user_channel_id_str = item
+                            .key
+                            .trim_start_matches(CHANNEL_CLOSURE_PREFIX)
+                            .splitn(2, '_') // Channel closures have `_{node_id}` at the end
+                            .collect::<Vec<&str>>()[0];
+                        let user_channel_id: [u8; 16] = FromHex::from_hex(user_channel_id_str)?;
+                        closure.user_channel_id = Some(user_channel_id);
+                    }
+                    activities.push(ActivityItem::ChannelClosed(closure));
+                }
```

Comment:
We have `list_channel_closures` which is now no longer used or accessed, we should move the logic here into a method that lists channel closes which can be reusable, and remove the old one.